src/dns/aardvark.rs

```diff
@@ -160,25 +159,44 @@ impl Aardvark {
         Ok(())
     }
 
-    pub fn commit_entry(&mut self, entry: AardvarkEntry) -> Result<()> {
+    fn commit_entry(&self, entry: AardvarkEntry) -> Result<()> {
         let mut data: String;
         let path = Path::new(&self.config).join(entry.network_name);
         let file_exists = path.exists();
-        let mut file = OpenOptions::new().append(true).create(true).open(&path)?;
+        let mut file = OpenOptions::new()
```

Comment:
I think the main idea here was to increase performance by only appending. I see the you fix this dual stack bug but IMO it must be fixed already before that. IPv4 and v6 should be submitted at once.

If I look at https://github.com/containers/aardvark-dns/blob/3baa1d79c19c6def206e960a294093ef10f3cc1e/src/test/config/podman_v6_entries/podman_v6_entries this config file is clearly wrong.
It should be something like this if I understand the format correctly
```
10.89.0.1, fdfd:733b:dc3:220b::3
7b46c7ad93fcbcb945c35286a5ba19d6976093e2ce39d2cb38ba1eba636404ab 10.89.0.2 fdfd:733b:dc3:220b::2 test1,7b46c7ad93fc
88dde8a2489780d3c8c90db54a9a97faf5dbe4f555b23e27880ca189dae0e2b0 10.89.0.3 fdfd:733b:dc3:220b::3 test2,88dde8a24897
```

@flouthoc @mheon PTAL

Reply:
> I think the main idea here was to increase performance by only appending. I see the you fix this dual stack bug but IMO it must be fixed already before that. IPv4 and v6 should be submitted at once.

Yes we agreed upon appending only logic for performance purpose.

> If I look at https://github.com/containers/aardvark-dns/blob/3baa1d79c19c6def206e960a294093ef10f3cc1e/src/test/config/podman_v6_entries/podman_v6_entries this config file is clearly wrong.
It should be something like this if I understand the format correctly

@Luap99 Testfile here is only added for a unit test testing of lookup logic, first line ensures spinning server on both ipv6 and ipv4 but its not necessary to provide both of them however we don't spin any server during unit testing it only tests resolution logic so https://github.com/containers/aardvark-dns/blob/3baa1d79c19c6def206e960a294093ef10f3cc1e/src/test/config/podman_v6_entries/podman_v6_entries should not be considered as standard so I'll rephrase the filename or amend it to match standards but i think its still a valid file cause aardvark server would still spin up server on provided `v4` ip and could contain mixed entries of `v4 and v6` which will be served on resolution logic ( again it should not be treated as standard )

Reply:
@flouthoc This is not just an example file.
It is exactly what netavark creates for a dual stack network `podman network create --ipv6`
```
cat /run/containers/networks/aardvark-dns/podman1
10.89.0.1
338947753ff897448fa754e8926b4ea68daa03840d7df26c8e839704f3f8f745 10.89.0.2  romantic_jang,338947753ff8
338947753ff897448fa754e8926b4ea68daa03840d7df26c8e839704f3f8f745  fd3b:2fee:9907:c91e::2 romantic_jang,338947753ff8
```
This is exactly the issue @martinetd is trying to fix here. The full config generation is wrong. The ID/names should never be duplicated instaed both IPv4/v6 should be on the same line.

Reply:
Ok so I've updated this patch to check if file exists before appending the entry, so at that point we can just gather all gateways and write the first line first.
That adds an extra open() but I believe that's a cheap price to pay.
I agree grouping everything on a single line would be better, but that's not reasonable with the current syntax allowing multiple subnets. A better fix would change aardvark syntax to have a list of all IPs instead of ipv4/ipv6 as a csv, but that's more than I'd like to chew and it doesn't make that fix any less valid. Better solutions can be done later, perfect is the enemy of good and all that.

and if you'd let me be an old fart for a moment here, let me just say we're already rewriting the file on container shutdowns (can't remove a line without that), so doing it on startup isn't that much worse... And anyway rewriting the whole file or just appending literally have the same cost as long as the file is <4k (which most will be), so that's premature optimization to me.